packages/vscode-extension/src/utils/upgrade.ts

```diff
@@ -24,6 +24,11 @@ export class ExtensionUpgrade {
       syncedVersion === undefined ||
       versionUtil.compare(teamsToolkitVersion, syncedVersion) === 1
     ) {
+      // if syncedVersion is undefined, then it is not upgrade user
```

Comment:
If user didn't upgrade, then the global state of UserState.IsUpgrade is undefined. In this case, the telemetry property will be "". Is that expected?

Reply:
this will not happen. in if condition, we check **versionUtil.compare(teamsToolkitVersion, syncedVersion) === 1**, so only upgraded user will go into this if condition.

Reply:
Yes, as you said, only upgraded user will go into this if condition and set global state of IsUpgrade. For user who didn't upgrade, this global state is not set and when you send telemetry, the value is defined and property will be "". Actually for this condition, these events should also be treated as existed user. Maybe you can also use query statement to group them to existing user group.